pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -251,13 +251,13 @@ func (a *cpuAccumulator) isNUMANodeFree(numaID int) bool {
 // Returns true if the supplied socket is fully available in `a.details`.
 // "fully available" means that all the CPUs in it are free.
 func (a *cpuAccumulator) isSocketFree(socketID int) bool {
-	return a.details.CPUsInSockets(socketID).Size() == a.topo.CPUsPerSocket()
+	return a.details.CPUsInSockets(socketID).Size() == a.topo.CPUDetails.CPUsInSockets(socketID).Size()
 }
 
 // Returns true if the supplied core is fully available in `a.details`.
 // "fully available" means that all the CPUs in it are free.
 func (a *cpuAccumulator) isCoreFree(coreID int) bool {
-	return a.details.CPUsInCores(coreID).Size() == a.topo.CPUsPerCore()
+	return a.details.CPUsInCores(coreID).Size() == a.topo.CPUDetails.CPUsInCores(coreID).Size()
```

Comment:
considering the issue https://github.com/kubernetes/kubernetes/issues/124066 as example, with this change the offlined core (id=25) won't  be used, right?

Reply:
1. Yes, the CPU(id=25) would not be used by container allocation because Core(id=25) has already allocated CPU(id=25) in previous Reserved Core allocation process. The buggy part is in `acc.takeFullCores` invoked by `takeByTopologyNUMAPacked` during container allocation. It thought Core(id=25) was free before this fix and used CPU(id=25),CPU(id=73) directly by refering raw Topo.
2. the core containing offline CPU(id=95) is not id=25, but id=47. it just affects `isCoreFree`. the offlined CPU(id=95) is never used because it does not appears in Topo(discovered by advisor) anyway.

Reply:
ok, thanks